src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,88 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.ValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private RegistrationService registrationService;
+
+    @BeforeAll
+    public static void setUpAll() {
+        final StorageDao storage = new StorageDaoImpl();
+        User existingUser = new User();
+        existingUser.setAge(20);
+        existingUser.setLogin("existingUser");
+        existingUser.setPassword("password123");
+        storage.add(existingUser);
+    }
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
```

Comment:
* why do you need a new `registrationService` before each test?
*  your user that you are creating in each test should be a lass field to create and initialize it in the `BeforeEach` and called a `validUser`, then you just set a single field that he needs for specific tests
* Existing user on the other hand should be only in the duplicate login test, since it is used only there
* magic values should be constants